patches/kaggle_bigquery.R

```diff
@@ -0,0 +1,43 @@
+# This library adds support for BigQuery (via the bigrquery library), by using
+# Kaggle's UserSecrets service to retrieve an OAuth access token for the connected
+# credentials attached to the running Kernel.
+#
+# Sample user code:
+#
+# library(bigrquery)
+# set_access_cred(TokenBigQueryKernel$new())
+# project <- "yes-theory-1" # put your project ID here
+# sql <- "SELECT year, month, day, weight_pounds FROM [publicdata:samples.natality] LIMIT 5"
+# query_exec(sql, project = project)
+
+KAGGLE_USER_SECRETS_TOKEN <- Sys.getenv("KAGGLE_USER_SECRETS_TOKEN")
+KAGGLE_BASE_URL <- Sys.getenv("KAGGLE_BASE_URL")
+GET_USER_SECRET_ENDPONT = "/requests/GetUserSecretRequest"
+
+library(httr)
+TokenBigQueryKernel <- R6::R6Class("TokenBigQueryKernel", inherit = Token2.0, list(

```

Comment:
Can you link to the `bigrquery` documentation section about the interface this class is implementing.